cwms-data-api/src/test/java/cwms/cda/data/dao/rsql/RSQLConditionBuilderTest.java

```diff
@@ -0,0 +1,187 @@
+package cwms.cda.data.dao.rsql;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.jooq.Condition;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import usace.cwms.db.jooq.codegen.tables.AV_TSV_DQU;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+class RSQLConditionBuilderTest {
+
+    @BeforeAll
+    static void forceUtc() {
+        // When jOOQ converts the Timestamp to a str it seems to use the system default tz.
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+        System.setProperty("user.timezone", "UTC");
+    }
+
+    @Test
+    void testBuildCondition() {
+        // Create a field resolver using the JooqFieldResolver
+        JooqFieldResolver resolver = new JooqFieldResolver(AV_TSV_DQU.AV_TSV_DQU);
+        RSQLConditionBuilder builder = new RSQLConditionBuilder(resolver);
+
+        Condition condition = builder.buildCondition("unit_id==EN;value>25");
+
+        assertNotNull(condition);
+        String conditionString = condition.toString();
+
+        assertTrue(conditionString.contains("\"AV_TSV_DQU\".\"UNIT_ID\" = 'EN'"));
```

Comment:
do these end up as bind variables by the time it gets to the database?

Reply:
Based on the visitor above it seems like they should, so I'm surprised you can test this way, unless it's just part of the `Condition::toString`?

Reply:
This test is relying on jOOQs Condition::toString.  Its not actually talking to the database, just using the view that it knows about from the jOOQ codegen.   I have two "Resolvers" that take the RSQL selectors and connect them to jOOQ fields.   I was hoping the JooqFieldResolver used in this test was going to make the timeseries impl really nice and clean.  It works when the jOOQ "from" clause is a simple view.  In reality the timeseries query is more complicated and I ended up using a different Resolver in the actual Dao code.  The JooqFieldResolver does work and makes the unit test cleaner.

Reply:
This example is operating on AV_TSV_DQU.UNIT_ID from the view.    JooqFieldResolver can find unit_id b/c it looks at all the table/view fields.    The Resolver I used in the actual Dao impl only knows about the fields you tell it about and I didn't include UNIT_ID b/c I figured it didn't make real-world sense.  Why would people give the end-point a timeseries_id and then ask it to filter by unit_id.  Didn't make real-world sense to me.   But RSQLConditionBuilder is written to potentially be more generic than just the timeseries usage so I just left the test in there.  It demonstrates that the RSQL syntax can have String comparisons.